### ISSUE TITLE
No start up after upgrade 

### ISSUE THREAD

**Issue by Dav0815** [nfarina/homebridge#1704]
### Expected Behavior

Was working fine before upgrade.

### Current Behavior

After performing an upgrade to the latest version the Homebridge service is hanging in an endless restart loop throwing errors.

### Homebridge Log / Command Output

```
Jan 14 17:07:05 hal node[24081]: *** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
Jan 14 17:07:05 hal node[24081]: *** WARNING *** Please fix your application to use the native API of Avahi!
Jan 14 17:07:05 hal node[24081]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
Jan 14 17:07:05 hal node[24081]: *** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
Jan 14 17:07:05 hal node[24081]: *** WARNING *** Please fix your application to use the native API of Avahi!
Jan 14 17:07:05 hal node[24081]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
Jan 14 17:07:05 hal homebridge[24081]: *** WARNING *** Please fix your application to use the native API of Avahi!
Jan 14 17:07:05 hal homebridge[24081]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
Jan 14 17:07:05 hal homebridge[24081]: *** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
Jan 14 17:07:05 hal homebridge[24081]: *** WARNING *** Please fix your application to use the native API of Avahi!
Jan 14 17:07:05 hal homebridge[24081]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
Jan 14 17:07:06 hal homebridge[24081]: /usr/lib/node_modules/hap-nodejs/node_modules/bindings/bindings.js:91
Jan 14 17:07:06 hal homebridge[24081]: throw err
Jan 14 17:07:06 hal homebridge[24081]: ^
Jan 14 17:07:06 hal homebridge[24081]: Error: Could not locate the bindings file. Tried:
Jan 14 17:07:06 hal homebridge[24081]: → /usr/lib/node_modules/hap-nodejs/node_modules/ed25519/build/ed25519.node
Jan 14 17:07:06 hal homebridge[24081]: → /usr/lib/node_modules/hap-nodejs/node_modules/ed25519/build/Debug/ed25519.node
Jan 14 17:07:06 hal homebridge[24081]: → /usr/lib/node_modules/hap-nodejs/node_modules/ed25519/build/Release/ed25519.node
Jan 14 17:07:06 hal homebridge[24081]: → /usr/lib/node_modules/hap-nodejs/node_modules/ed25519/out/Debug/ed25519.node
Jan 14 17:07:06 hal homebridge[24081]: → /usr/lib/node_modules/hap-nodejs/node_modules/ed25519/Debug/ed25519.node
Jan 14 17:07:06 hal homebridge[24081]: → /usr/lib/node_modules/hap-nodejs/node_modules/ed25519/out/Release/ed25519.node
Jan 14 17:07:06 hal homebridge[24081]: → /usr/lib/node_modules/hap-nodejs/node_modules/ed25519/Release/ed25519.node
Jan 14 17:07:06 hal homebridge[24081]: → /usr/lib/node_modules/hap-nodejs/node_modules/ed25519/build/default/ed25519.node
Jan 14 17:07:06 hal homebridge[24081]: → /usr/lib/node_modules/hap-nodejs/node_modules/ed25519/compiled/6.12.3/linux/arm/ed25519.node
Jan 14 17:07:06 hal homebridge[24081]: at bindings (/usr/lib/node_modules/hap-nodejs/node_modules/bindings/bindings.js:88:9)
Jan 14 17:07:06 hal homebridge[24081]: at Object.<anonymous> (/usr/lib/node_modules/hap-nodejs/node_modules/ed25519/index.js:1:99)
Jan 14 17:07:06 hal homebridge[24081]: at Module._compile (module.js:570:32)
Jan 14 17:07:06 hal homebridge[24081]: at Object.Module._extensions..js (module.js:579:10)
Jan 14 17:07:06 hal homebridge[24081]: at Module.load (module.js:487:32)
Jan 14 17:07:06 hal homebridge[24081]: at tryModuleLoad (module.js:446:12)
Jan 14 17:07:06 hal homebridge[24081]: at Function.Module._load (module.js:438:3)
Jan 14 17:07:06 hal homebridge[24081]: at Module.require (module.js:497:17)
Jan 14 17:07:06 hal homebridge[24081]: at require (internal/module.js:20:19)
Jan 14 17:07:06 hal homebridge[24081]: at Object.<anonymous> (/usr/lib/node_modules/hap-nodejs/lib/HAPServer.js:8:15)

```

**Comment by xena68** [comment]
Having a similar problem.  Upgraded Node, homebridge, node-gyp and am getting this error:

`$ homebridge
/usr/local/lib/node_modules/homebridge/node_modules/bindings/bindings.js:96
  throw err
  ^

Error: Could not locate the bindings file. Tried:
 â /usr/local/lib/node_modules/homebridge/node_modules/ed25519-hap/build/ed25519.node
 â /usr/local/lib/node_modules/homebridge/node_modules/ed25519-hap/build/Debug/ed25519.node
 â /usr/local/lib/node_modules/homebridge/node_modules/ed25519-hap/build/Release/ed25519.node
 â /usr/local/lib/node_modules/homebridge/node_modules/ed25519-hap/out/Debug/ed25519.node
 â /usr/local/lib/node_modules/homebridge/node_modules/ed25519-hap/Debug/ed25519.node
 â /usr/local/lib/node_modules/homebridge/node_modules/ed25519-hap/out/Release/ed25519.node
 â /usr/local/lib/node_modules/homebridge/node_modules/ed25519-hap/Release/ed25519.node
 â /usr/local/lib/node_modules/homebridge/node_modules/ed25519-hap/build/default/ed25519.node
 â /usr/local/lib/node_modules/homebridge/node_modules/ed25519-hap/compiled/8.11.1/linux/x64/ed25519.node
    at bindings (/usr/local/lib/node_modules/homebridge/node_modules/bindings/bindings.js:93:9)
    at Object.<anonymous> (/usr/local/lib/node_modules/homebridge/node_modules/ed25519-hap/index.js:1:99)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/HAPServer.js:8:15)
`
I went to the bindings folder and tried to follow the above recommendation:

`$ sudo node-gyp BUILDTYPE=Release rebuild                      gyp info it worked if it ends with ok
gyp info using node-gyp@3.6.2
gyp info using node@8.10.0 | linux | x64
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/usr/local/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/homebridge/node_modules/bindings/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/root/.node-gyp/8.10.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/root/.node-gyp/8.10.0',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/root/.node-gyp/8.10.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/usr/local/lib/node_modules/homebridge/node_modules/bindings',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp: binding.gyp not found (cwd: /usr/local/lib/node_modules/homebridge/node_modules/bindings) while trying to load binding.gyp
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/node-gyp/lib/configure.js:336:16)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Linux 3.16.0-4-amd64
gyp ERR! command "/usr/local/bin/node" "/usr/local/bin/node-gyp" "BUILDTYPE=Release" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/homebridge/node_modules/bindings
gyp ERR! node -v v8.10.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
`
Any suggestions?

**Comment by KhaosT** [comment]
Can you guys try `sudo npm -g uninstall homebridge` and then `npm -g install homebridge --unsafe-perm`?

**Comment by iandronowicz** [comment]
You have to do the gyp rebuild in the /usr/local/lib/node_modules/homebridge/node_modules/ed25519-hap/ folder

**Comment by xena68** [comment]
@iandronowicz Tried that, but didn't work.

@KhaosT uninstll and then install per the above worked for me.  As always,  **Thank You** for this awesome tool and your awesome support!

**Comment by Dav0815** [comment]
Sorry for not sending an update earlier, but at the point when I had the issue, I fixed the problem by a downgrade by one version and a second update attempt. I dont know why, but it seems that my first update did not cover all the libs, which is why I ran into trouble. 

Also, did you update any other component ? I had times, where home assistant and appdaemon would require different libraries. I guess that is one of the reasons to use docker.
